src/cli/mod.rs

```diff
@@ -564,6 +564,45 @@ async fn push_new_release(
     span.record("source", tracing::field::display(source.clone().display()));
     tracing::debug!("Found source");
 
+    if flake_dir.join("flake.lock").exists() {
+        let output = tokio::process::Command::new("nix")
+            .arg("flake")
+            .arg("metadata")
+            .arg("--json")
+            .arg("--no-update-lock-file")
```

Comment:
This is the special sauce right here: if `nix flake metadata --json` would have updated the flake.lock, this causes Nix (and thus the entire upload) to fail (which we want).